p2p/conn/connection.go

```diff
@@ -761,24 +756,36 @@ func (ch *Channel) updateStats() {
 	ch.recentlySent = int64(float64(ch.recentlySent) * 0.8)
 }
 
-//-----------------------------------------------------------------------------
+//----------------------------------------
+// Packet
 
-const (
-	defaultMaxMsgPacketPayloadSize = 1024
+type Packet interface {
+	AssertIsPacket()
```

Comment:
Otherwise anything can be a Packet.  This is just to leverage the Golang typechecker.  It's compile time assertion, or basically just type-checking.  Arguably should be called "TypeCheckIsPacket", but we already have a convention of using "AssertIs"